apis/common/v1/connection_details.go

```diff
@@ -148,25 +160,44 @@ type VaultAuthKubernetesConfig struct {
 type VaultAuthConfig struct {
 	// Method configures which auth method will be used.
 	Method VaultAuthMethod `json:"method"`
+	// Token configures Token Auth for Vault.
+	// +optional
+	Token *VaultAuthTokenConfig `json:"token,omitempty"`
 	// Kubernetes configures Kubernetes Auth for Vault.
 	// +optional
 	Kubernetes *VaultAuthKubernetesConfig `json:"kubernetes,omitempty"`
 }
 
+// VaultKVVersion represent API version of the Vault KV engine
+// https://www.vaultproject.io/docs/secrets/kv
+type VaultKVVersion string
+
+const (
+	// VaultKVVersionV1 indicates that "Kubernetes Auth" will be used to
+	// authenticate to Vault.
+	// https://www.vaultproject.io/docs/auth/kubernetes
+	VaultKVVersionV1 VaultKVVersion = "v1"
+
+	// VaultKVVersionV2 indicates that "Token Auth" will be used to
+	// authenticate to Vault.
+	// https://www.vaultproject.io/docs/auth/token
+	VaultKVVersionV2 VaultKVVersion = "v2"
+)
+
 // VaultSecretStoreConfig represents the required configuration for a Vault
 // secret store.
 type VaultSecretStoreConfig struct {
 	// Server is the url of the Vault server, e.g. "https://vault.acme.org"
 	Server string `json:"server"`
 
-	// ParentPath is the path to be prepended to all secrets.
-	ParentPath string `json:"parentPath"`
+	// MountPath is the mount path of the KV secrets engine.
+	MountPath string `json:"mountPath"`
```

Comment:
Each secret engine in Vault is mounted to a specific path and yes this is well-known terminology. For example: https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/mount#path

Initially, I was thinking we don't have to know the mount point as long as we know the parent path of the secret but it turned out to we need to know the exact mount point to work with [kv v2 engine](https://www.vaultproject.io/docs/secrets/kv/kv-v2) because we need to append [data/](https://www.vaultproject.io/api-docs/secret/kv/kv-v2#create-update-secret) or [metadata/](https://www.vaultproject.io/api-docs/secret/kv/kv-v2#create-update-metadata) right after mount point. And there is no guarantee that mount point will be first element in the parent path.

As an example, one can mount kv v2 secret engine at `engines/kv/team-a` and may want to store crossplane secrets under `engines/kv/team-a/infrastructure`. If we get the latter as `parentPath`, there is no way for us to figure out whether mount point is `engines/` or `engines/kv` or   `engines/kv/team-a` ... and we need to make request to `engines/kv/team-a/data/infrastructure/<secret-name>` and `engines/kv/team-a/metadata/infrastructure/<secret-name>`. 

One limitation that I am just realizing is it wouldn't be possible to configure the parent path of the secret other than `defaultScope` in top-level config and namespacing. We may still want to consider adding an optional `parentPath` variable in addition to `mountPath` but would prefer to do so when someone requests.